demonstrations/tutorial_learningshallow.py

```diff
@@ -0,0 +1,433 @@
+r"""Learning shallow quantum circuits with local inversions and circuit sewing
+==============================================================================
+
+In a recent paper `Learning shallow quantum circuits <https://arxiv.org/abs/2401.10095>`_ [#Huang]_, Huang et al
+introduce and prove performance bounds on efficient algorithms to learn constant depth circuits.
+At the heart of the paper lie local inversions that locally undo a quantum circuit, as well as a circuit "sewing"
+technique that let's one construct a global inversion from those.
+We are going to review these new concepts and showcase them with an implementation in PennyLane.
+
+Introduction
+------------
+
+Shallow, constant depth quantum circuit are provably powerful [#Bravyi]_.
+At the same time, quantum neural networks are known to be difficult to train [#Anschuetz]_.
```

Comment:
This is a very sweeping statement, we train them all the time ;) I am not sure if a reference to QNNs is needed here, since it insinuates a non-existing QML flavour? 

Rather say:

```suggestion
At the same time, they lead to optimization problems whose landscape has many local minima [#Anschuetz]_.
```

Not ideal yet though...maybe move or remove that reference here?

Reply:
Maybe this?
```suggestion
At the same time, training quantum neural networks, just like their classical counterparts, are non-trivial to optimize [#Anschuetz]_.
```

Reply:
Actually, the statement as such wasn't my concern, but that the logical flow is unclear: a) shallow circuits are good, b) QNNs are difficult to train, c) the paper asks how shallow circuits can be trained -> what have the QNNs to do with this?

In order to not explain this complicated relationship I'd just remove the QNNs!

Reply:
I am not sure I understand 😅 

To me the hook of the paper is a) shallow circuits are powerful b) training in general is hard, also constant-depth ones, c) here is a version to train them that is provably not hard.

Is the problem that I am using qnn and circuit interchangably?

Reply:
I just rephrased the sentence now to "Shallow, constant depth quantum circuit are provably powerful [#Bravyi]_.
At the same time, they are known to be difficult to train [#Anschuetz]_."

Reply:
Yes! That's perfect!